### DISCUSSION TITLE
Why are the Micronaut Java generators still in beta according to OpenAPI Generator?

### DISCUSSION THREAD

**Discussion by runenielsen** [micronaut-projects/micronaut-openapi#1161] [General]
Hi.

I was wondering why the Micronaut client and server Java generators are still marked as being in beta by OpenAPI Generator, seeing as this project (micronaut-openapi) is 5 years old, and doesn't seem to be in beta?

https://openapi-generator.tech/docs/generators

I assume you use these two generators under the hood... Or is that a misunderstanding?

Thanks in advance 🙂
Rune

**Comment by dstepanov** [comment]
@andriy-dmytruk Do you know?

**Comment by andriy-dmytruk** [comment]
Not sure. But the generator that is part of the openapi-generator project is using 3.0.x Micronaut, while the generation of Micronaut 4.0.x was moved to this repository.

It is now supported as part of Micronaut projects along with maven and gradle plugins that help integrate generation into the build.

To clarify, this project initially generated the spec from micronaut annotations and didn't generate micronaut code from OpenAPI spec. Since Micronaut 4.0.0 it does both.

**Reply by runenielsen** [reply]
Thanks for the answer 🙂

**Comment by scprek** [comment]
Additionally, the https://openapi-generator.tech/docs/generators mention no support for kotlin, but it seems that the microanut-gradle-plugin in fact supports kotlin.
